### PR TITLE
[POC] Using microlink cards to generate og images

### LINKED ISSUE #491: Generate social images on runtime in the server
**Is your feature request related to a problem? Please describe.**
Right now we need to keep every possible social image in the repos of the website, eventually this make the repo heavier due the big amount of images.

**Describe the solution you'd like**
Generate a Node.js function exposing [@contentz-tech/social](https://github.com/contentz-tech/social) using HTTP, it should be used like:

```http
GET /api/social?title=My article title&description=My article description
```

This API endpoint should reply with the generate image, not keeping it on disk, and with a Cache-Control header to ensure proper CDN y browser cache, since the title and description are in the URL any update on those will invalidate the social image.

---

From the HTML side, we will add the `<meta>` tag pointing to that endpoint and with the page title and description. For the special cases (home, CV, articles, links, error) we can generate the same tag but with the title and description based on what we want to generate, this will simplify lots of code too.

### PR DESCRIPTION
**Pull request by cristianbgp**
Using [Microlink Cards with contentz preset](https://cards.microlink.io/editor?preset=contentz) to generate og images.

Might need a little more testing if everything is ok. Also, this might deprecate the use of @contentz/social

Resolves #491

### CODE REVIEW

**Comment by cristianbgp** [comment]
I used ngrok to test it, it seems that is working well 
image

**Comment by sergiodxa** [comment]
Could you update the E2E tests to test the meta tag for the og in the HTML of each type of page?

**Comment by cristianbgp** [comment]
@sergiodxa I've just added those tests :eyes:

File: end-to-end.test.js [hunk 1]
```diff
@@ -45,6 +60,24 @@ describe("End to End", () => {
     await expect(page).toMatch("Go to the list of Articles");
     await expect(page).toMatch("Go to the list of Talks");
     await expect(page).toMatch("Go to the list of Shared Links");
+    const ogURLContent = await page.$eval(
```
Comment (sergiodxa) [inline end-to-end.test.js:63]: You can extract this to a function outside the test and reuse it passing the page

Comment (cristianbgp) [reply end-to-end.test.js:63]: For only the part of getting the content of the meta tag?

Comment (cristianbgp) [reply end-to-end.test.js:63]:
```javascript
async function getMetaTagContent(page, property) {
  return await page.$eval(
    `meta[property='${property}']`,
    (element) => element.content
  );
}
```

File: end-to-end.test.js [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
+
 describe("End to End", () => {
   test("Home Page", async () => {
     await page.goto("http://localhost:5000/");
     await expect(page).toMatch("Contentz Test");
     await expect(page).toMatch(
       "Create Content, Get a Highly Optimized Website"
     );
+    const ogURLContent = await getMetaTagContent(page, "og:url");
+    expect(ogURLContent).toMatch("https://contentz.tech/home/");
```
Comment (sergiodxa) [inline end-to-end.test.js:16]: The OG url is wrong I think? It should be / not /home, right?

Comment (cristianbgp) [reply end-to-end.test.js:16]: 😮 That was how it was working before, I'm going to change it

**Comment by sergiodxa** [comment]
:shipit:

**Pull request merged**